projecttacoma/measure-repository
# Replace <resourceType>/<id> links with info cards

# Summary
Replaces `<resourceType>/<id>` links with info cards for the resources available on the server. Each info card links to `[resourceType]/[id].tsx`.

## New Behavior
When browsing the available resources on the measure repository service, the user can browse info cards containing information about the resource beyond just its resource type and id. Clicking on a resource card opens it with more information, and a “view” button. Clicking the “view” button on a resource card will bring the user to the next page with the JSON/CQL/narrative content.

## Code Changes
* `ResourceInfoCard.tsx` - styling for “info cards” containing information about each resource. This includes `id`, `identifier`, `url`, `version`, `name`, and `status`. These fields are only included on the card if they are defined
* `[resourceType].tsx` - changes to server side props to get all the info needed for the info card. Update logic that loops over all the resources, replacing the existing `Link`/`Button` code with the `ResourceInfoCard`s
* `fhir.ts` - New interface for capturing all the resource info card information

# Testing Guidance
* Upload resources to the measure repository service - can use the `db:loadBundle` script to accomplish this
* Once the database is populated, use the resource list on the left-hand side to select a resource type —> this will bring you to `[resourceType].tsx`
* Select a resource info card by clicking on it 
* Inspect the style of the resource info card — sizing, color scheme, border color upon selection, order of the different fields, etc.
* Check that "viewing" a resource via the "view" button brings the user to appropriate page with the JSON/CQL/narrative content
* Double-check the typing of the `ResourceInfo` interface

Open Questions and Areas for Feedback:
* Thoughts on the card format? Should we have all the info appear on each card, or should we keep the current functionality, which only shows fields like name/version/status if the resource is selected? Trying to find a balance between how much info to show on the page at a time (as to not overwhelm the user) and aesthetics/scalability (since this is a demo app and I imagine we won’t have hundreds of resources on the server at a time)
    * If we want to display all the info at once, we could consider removing the “view” button entirely, and embed the link directly in the resource info card
* Currently, the identifier is only displayed on the card if both the `system` and `value` are defined. Open to thoughts on the best way to handle this, as `system` and `value` both have cardinality 0..1 (link to [spec](https://build.fhir.org/datatypes.html#Identifier) with more details)
